### PR TITLE
ALTAPPS-1028: Android share streak

### PR DESCRIPTION
**Pull request by XanderZhu**
**YouTrack Issues**:
[#ALTAPPS-1028](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-1028)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [x] All checks have been passed;
- [x] Changes have been checked locally.

**Description**
Support image and text sharing for streak

### CODE REVIEW

File: androidHyperskillApp/src/main/java/org/hyperskill/app/android/step/view/delegate/StepDelegate.kt [hunk 1]
```diff
@@ -82,11 +90,22 @@ class StepDelegate<TFragment>(
...
+                            .showIfNotExists(fragment.childFragmentManager, ShareStreakDialogFragment.TAG)
                     }
                     is StepCompletionFeature.Action.ViewAction.ShowShareStreakSystemModal -> {
-                        // TODO: ALTAPPS-1028 Show system share streak modal (after "Share your streak" button clicked)
-                        // on the problem of day solved modal
+                        val shareIntent = ShareUtils.getShareDrawableIntent(
+                            fragment.requireContext(),
+                            getShareStreakDrawableRes(stepCompletionAction.streak),
+                            text = getShareStreakText(fragment.requireContext()),
+                            title = "Share your streak"
```
Comment (ivan-magda) [inline androidHyperskillApp/src/main/java/org/hyperskill/app/android/step/view/delegate/StepDelegate.kt:102]: You can use shared resource `share_streak_modal_title`

**Review by ivan-magda — COMMENTED** [review]
One missed thing - we are also showing streak sharing info after completion problem of the day.

See for more information:
- `StepCompletionFeature.Action.ViewAction.ShowProblemOfDaySolvedModal`
- `StepCompletionFeature.Message.ProblemOfDaySolvedModalShareStreakClicked`

_A tiny suggestion, for "Share you streak button" it's better to use violet outline button style, as we did before many times with the other modals._

Screenshot 2023-11-15 at 7 55 47 PM

File: androidHyperskillApp/src/main/java/org/hyperskill/app/android/step/view/delegate/StepDelegate.kt [hunk 2]
```diff
@@ -121,4 +133,36 @@ class StepDelegate<TFragment>(
                 .checkNotificationChannelAvailability(context, HyperskillNotificationChannel.DailyReminder)
         }
     }
+
+    @DrawableRes
+    private fun getShareStreakDrawableRes(streak: Int): Int =
```
Comment (ivan-magda) [inline androidHyperskillApp/src/main/java/org/hyperskill/app/android/step/view/delegate/StepDelegate.kt:138]: `ShareStreakDialogFragment` also has the same function `getImageRes`, maybe it's better to provide image resource for `ShareStreakDialogFragment` by argument

File: gradle/app.versions.toml [hunk 3]
```diff
@@ -2,5 +2,5 @@
 minSdk = '24'
 targetSdk = '33'
 compileSdk = '33'
-versionName = '1.41'
-versionCode = '233'
\ No newline at end of file
+versionName = '1.42'
```
Comment (ivan-magda) [inline gradle/app.versions.toml:5]: Hm, strange why we have here changes from the current release branch `release/1.42`

**Review by ivan-magda — APPROVED** [review]

File: gradle/app.versions.toml (hunk 3 above)
Comment (XanderZhu) [reply gradle/app.versions.toml:5]: I accidentally checked out this branch from release branch

**Pull request merged**